### PR TITLE
Loki mixin: publish compiled version of the mixin

### LINKED ISSUE #6234: operation: Make `loki-mixin` available to non-jsonnet users as well.
Currently Loki operators who are using jsonnet can only benefit from dashboards and alerts from [loki-mixin](https://github.com/grafana/loki/tree/main/production/loki-mixin). We should support those dashboards and alerts for non jsonnet users as well.

Something similar to what mimir does with [mimir-mixin-compiled.](https://github.com/grafana/mimir/tree/main/operations/mimir-mixin-compiled)

### PR DESCRIPTION
**Pull request by pracucci**
**What this PR does / why we need it**:
I'm leading an effort to publish and document dashboards, alerts and runbooks across Mimir, Loki and Tempo. In this PR I want to propose to publish a compiled version of the mixin (with default config), following the same pattern used by Mimir.

_Documentation is out of the scope of this PR. Will be done in a follow up PR._

**Which issue(s) this PR fixes**:
Partially addresses #6234 (doc still missing)

**Special notes for your reviewer**:
N/A

**Checklist**
- [ ] Documentation added
- [ ] Tests updated
- [ ] Is this an important fix or new feature? Add an entry in the `CHANGELOG.md`.
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -27,7 +27,7 @@ DOCKER_IMAGE_DIRS := $(patsubst %/Dockerfile,%,$(DOCKERFILES))
 BUILD_IN_CONTAINER ?= true
 
 # ensure you run `make drone` after changing this
-BUILD_IMAGE_VERSION := 0.20.4
+BUILD_IMAGE_VERSION := publish-compiled-mixin-a795b93-WIP
```
Comment (kavirajk) [inline Makefile:30]: @pracucci I don't if  WIP suffix is intentional? I think WIP suffix comes from the fact, you may have some uncommitted changes while building image via `make loki-build-image`.

We can get rid of it by removing all git modified files in local workspace.

Can we keep some static tag instead? (e.g: `0.20.5`)

File: Makefile [hunk 2]
```diff
@@ -210,6 +210,28 @@ clients/cmd/promtail/promtail-debug:
...
+		-v $(shell pwd):/src/loki$(MOUNT_FLAGS) \
+		$(IMAGE_PREFIX)/loki-build-image:$(BUILD_IMAGE_VERSION) $@;
+else
+	@rm -rf $(MIXIN_OUT_PATH) && mkdir $(MIXIN_OUT_PATH)
+	@cd $(MIXIN_PATH) && jb install
+	@mixtool generate all --output-alerts $(MIXIN_OUT_PATH)/alerts.yaml --output-rules $(MIXIN_OUT_PATH)/rules.yaml --directory $(MIXIN_OUT_PATH)/dashboards ${MIXIN_PATH}/mixin.libsonnet
+endif
+
+loki-mixin-check: loki-mixin
+	@echo "Checking diff:"
```
Comment (kavirajk) [inline Makefile:232]:
```suggestion
	@echo "Checking diff"
```
nit: We can remove `:` next git diff follows in the next line anyway.

It looks something like this now.
```
loki$ make loki-mixin-check 
docker run --rm --tty -i \
	-v /home/kavirajk/src/loki:/src/loki:delegated \
	grafana/loki-build-image:publish-compiled-mixin-a795b93-WIP loki-mixin;
make: Entering directory '/src/loki'
make: Leaving directory '/src/loki'
Checking diff:
diff --git a/production/loki-mixin-compiled/something.txt b/production/loki-mixin-compiled/something.txt
deleted file mode 100644
index e69de29bb..000000000
Please build mixin by running 'make loki-mixin'
make: *** [Makefile:233: loki-mixin-check] Error 1

```

**Review by kavirajk — APPROVED** [review]
This is super nice :heart: Thanks @pracucci. Left few small nits. Overall looks great.

File: Makefile (hunk 1 above)
Comment (pracucci) [reply Makefile:30]: Yeah, see PR description:

> Draft PR to get a feedback. Then I will open a dedicated PR to properly update the build image.

Given you're approving this PR, I'm now going to open a PR to publish 0.20.5 (following the 2 steps process in your instructions).

Comment (pracucci) [reply Makefile:30]: See: https://github.com/grafana/loki/pull/6255

**Pull request merged**